components/todo-input.js

```diff
@@ -1,20 +1,96 @@
-import React from "react";
-import styled from "styled-components";
+import React from 'react';
+import styled from 'styled-components';
 
-export default function TodoInput() {
-  return (
-    <Wrapper>
-      리액트-투두
-      <br />
-      시간을 입력하는 칸
-      <br />
-      Todo를 입력하는 칸
-    </Wrapper>
-  );
+export default function TodoInput(props) {
+	const { date, todo, onChangeTodo, onChangeDate, onSubmit } = props;
+
+	const maxLengthCheck = (object) => {
+		if (object.value.length > object.maxLength) {
+			object.value = object.value.slice(0, object.maxLength);
+		}
+	};
+
+	return (
+		<Form>
+			<InputSection>
+				<P>시간</P>
+				<Input
+					maxlength="8"
+					oninput="maxLengthCheck(this)"
+					type="number"
+					placeholder="날짜를 입력하세요 (ex.20200404)"
+					value={date}
+					onChange={onChangeDate}
+				/>
+			</InputSection>
+
+			<InputSection>
+				<P>TODO</P>
+				<TextArea
+					placeholder="할 일을 입력하세요"
+					value={todo}
+					onChange={onChangeTodo}
+				/>
+			</InputSection>
+			<Button onClick={onSubmit}>등록</Button>
+		</Form>
+	);
 }
 
-const Wrapper = styled.div`
-  border: solid 1px;
-  font-size: 18px;
-  flex: 1;
+const Input = styled.input`
+	width: 80%;
+	border-width: 1px;
+	border-style: solid;
+	border-color: rgb(97, 97, 97);
+	border-image: initial;
+	padding: 0.5rem 0.8rem;
+`;
+const TextArea = styled.textarea`
+	width: 80%;
+	height: 30vh;
+	resize: none;
+	border-width: 1px;
+	border-style: solid;
+	border-color: rgb(97, 97, 97);
+	padding: 0.5rem 0.8rem;
+`;
+const P = styled.p`
+	font-size: 1.5rem;
+	color: white;
+	margin: 0;
+`;
+const InputSection = styled.div`
+	color: white;
+	width: 100%;
+	display: flex;
+	flex-direction: row;
+	-webkit-box-pack: justify;
+	justify-content: space-between;
+	align-items: flex-start;
+	margin-bottom: 3rem;
+`;
+const Form = styled.form`
+	font-size: 18px;
+
+	width: 100%;
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+	height: 37rem;
+	flex: 1;
+	border-width: 1px;
+	border-style: solid;
+	padding: 1rem 2rem;
+`;
```

Comment:
```suggestion
const Wrapper = styled.form`
	font-size: 18px;
	width: 100%;
	display: flex;
	flex-direction: column;
	align-items: center;
	height: 37rem;
	border: 1px solid black;
	padding: 1rem 2rem;
`;
```
1. 가장 바깥쪽 태그 이름은 ```Wrapper```로 지어주시면 됩니다.
2. ```flex:1;``` 이 필요할까용?
3. border에 관한 코드는 ```border: 1px solid black;```으로 한 줄에 쓸 수 있습니다.